beetsplug/fetchart.py

```diff
@@ -230,22 +231,42 @@ def fetch_image(self, candidate, extra):
             with closing(self.request(candidate.url, stream=True,
                                       message=u'downloading image')) as resp:
                 ct = resp.headers.get('Content-Type', None)
-                if ct not in CONTENT_TYPES:
-                    self._log.debug(
-                        u'not a supported image: {}',
-                        resp.headers.get('Content-Type') or u'no content type',
-                    )
-                    candidate.path = None
-                    return
-
-                # Generate a temporary file with the correct extension.
-                with NamedTemporaryFile(suffix=b'.' + CONTENT_TYPES[ct][0],
-                                        delete=False) as fh:
-                    for chunk in resp.iter_content(chunk_size=1024):
+
+                # Generate a temporary file and guess the extension based on
+                # the Content-Type header. This may be wrong for badly
+                # configured servers. E.g. fanart.tv only allows .jpg uploads
+                # and ALWAYS returns a image/jpeg Content-Type, but other
+                # formats with a erroneous .jp(e)g extension apparently can
+                # sneak through the upload filter. Therefore validate the type
+                # using the file magic.
+                data = resp.iter_content(chunk_size=1024)
```

Comment:
Good catch, in fact I don't know as I've no real knowledge of the ``requests`` module. Will check that.